CDTDatastore/CDTReplicator/CDTReplicator.m

```diff
@@ -109,26 +110,51 @@ + (NSString *)stringForReplicatorState:(CDTReplicatorState)state
     }
 }
 
++ (CDTReplicator *)replicatorWithConfiguration:(CDTAbstractReplication *)configuration
```

Comment:
OK I see what you mean, this is a convenience for calling `alloc`, `initWithTDDatabaseManager`. But I wonder if we need it because it might distract/confuse users from doing it the "proper" way which is to use the `CDTReplicatorFactory`.